### DISCUSSION TITLE
When using the "ViewPort" scale option, have some way of showing art behind the viewport window.

### DISCUSSION THREAD

**Discussion by sysl-dev** [godotengine/godot-proposals#11209] [2D]
![th-290901894](https://github.com/user-attachments/assets/19afb1d6-6bc7-4c32-8bc3-e18802b9da44)

I'm thinking of games like Undertale as an example, where behind the game (which appears to be a viewport that scales to keep pixel sharpness) there is an image that can be changed out or set as the game goes on, instead of a straight black image.

Yes, this can be manually done without it, but you lose benefits of viewport scaling (like control nodes scaling matching pixel density without setting up your own sub-viewport for rendering)

This is not a huge deal, most people still have 16:9 screens, but as winder screens becomes more popular I would like to have options they could toggle instead of plain black.

Note: This is not a discussion about game ui and scaling, this is a feature for a type of game that is trying to emulate a retro style of display. (Gameboy jams being a great example, I know people would love to be able to put a fake gameboy-like behind gbjam entries)

**Comment by theraot** [comment]
See https://github.com/godotengine/godot-proposals/issues/10987 and the proposals linked there. In particular https://github.com/godotengine/godot-proposals/issues/2319.

**Comment by Calinou** [comment]
Thanks for the proposal! Consolidating in https://github.com/godotengine/godot-proposals/issues/2319.
